### ISSUE TITLE
Add methods to sourceCode to convert range indices to line/column locations

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#8073]
Previous discussion: https://github.com/eslint/eslint/pull/7645

A lot of whitespace-related rules such as `no-tabs` operate on the entire source text, rather than a particular node or token. Since the `context.report` API requires a line and a column, it can be tedious to report problems based on matching against the source text -- a rule typically has to implement custom line-splitting logic to convert a source text index into a (line, column) pair.

In `ast-utils.js`, we have helper functions to do this: `getLocationFromRangeIndex` and `getRangeIndexFromLocation`. I think it would be a good idea to add these functions to the `SourceCode.prototype` instead so they can be used by non-core rules.

```js
class SourceCode {
  // Converts a character index in a source file into a line, column pair
  getLocationFromRangeIndex(index: number): { line: number, column: number }

  // Converts a line, column pair into a character index in a file
  getRangeIndexFromLocation(loc: { line: number, column: number }): number
}
